lib/spack/spack/detection/path.py

```diff
@@ -252,11 +247,22 @@ def find(
         patterns = self.search_patterns(pkg=pkg_cls)
         if not patterns:
             return []
-        path_hints = self.path_hints(pkg=pkg_cls, initial_guess=initial_guess)
-        candidates = self.candidate_files(patterns=patterns, paths=path_hints)
+        if initial_guess is None:
+            initial_guess = spack.util.environment.get_path("PATH")
+            initial_guess.extend(self.path_hints(pkg_cls))
+        candidates = self.candidate_files(patterns=patterns, paths=initial_guess)
         result = self.detect_specs(pkg=pkg_cls, paths=candidates)
         return result
 
+    def __call__(self, *args, **kwargs):
+        """Need entire Finder object preserved in subprocesses
+        not just only what is required for a single method.
+
+        This call method allows for the class to be executed and forces
+        `spawn` to copy the entire class instance.
+        """
+        return self.find(*args, **kwargs)
```

Comment:
Can you expand on this? I don't quite understand the problem and how it is solved :thinking:

Reply:
Sorry, I should have taken this out. This was one of my earlier attempts to pass the monkeypatches into the subprocess. It's since been rendered totally unnecessary and I was planning on dropping it once we ok'd the new test I added and I went back to cleanup the history.